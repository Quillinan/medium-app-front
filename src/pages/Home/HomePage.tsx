import { useAuth } from '@auth/useAuth';
import MainContent from '@components/Content/MainContent/MainContent';
import Header from '@components/Header/Header';
import NavBar from '@components/NavBar/NavBar';
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

const HomePage: React.FC = () => {
  const { removeAuth } = useAuth();
  const navigate = useNavigate();
  const auth = sessionStorage.getItem('isAuthenticated');
  const [currentTab, setCurrentTab] = useState('Menu');

  const logout = () => {
    removeAuth();
    navigate('/login');
  };

  useEffect(() => {
    if (!auth) {
      logout();
    }
  }, [auth]);

  return (
    <div className='min-h-full'>
      <NavBar currentTab={currentTab} onTabChange={setCurrentTab} />
      <Header title={currentTab} />
      <MainContent currentTab={currentTab} />
    </div>
  );
};

export default HomePage;
