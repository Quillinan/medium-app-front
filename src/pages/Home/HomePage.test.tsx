import { fireEvent, render, screen } from '@testing-library/react';
import HomePage from './HomePage';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '@auth/useAuth';
import { Mock } from 'vitest';

vi.mock('@auth/useAuth');
vi.mock('react-router-dom', () => ({
  useNavigate: vi.fn(),
}));

describe('HomePage', () => {
  const mockRemoveAuth = vi.fn();
  const mockNavigate = vi.fn();

  beforeAll(() => {
    vi.spyOn(console, 'error').mockImplementation(() => {});
  });

  afterAll(() => {
    (console.error as Mock).mockRestore();
  });

  beforeEach(() => {
    (useAuth as Mock).mockReturnValue({ removeAuth: mockRemoveAuth });
    (useNavigate as Mock).mockReturnValue(mockNavigate);
    sessionStorage.setItem('isAuthenticated', 'true');
  });

  afterEach(() => {
    sessionStorage.clear();
    vi.clearAllMocks();
  });

  it('should logout and navigate to login when not authenticated', () => {
    sessionStorage.removeItem('isAuthenticated');
    render(<HomePage />);

    expect(mockRemoveAuth).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith('/login');
  });

  it('should render the login page with navbar, header, and main content', () => {
    render(<HomePage />);

    expect(screen.getByTestId('navbar')).toBeInTheDocument();
    expect(screen.getByTestId('header')).toBeInTheDocument();
    expect(screen.getByTestId('main-content')).toBeInTheDocument();
  });

  it('should change the current tab when a new tab is clicked', () => {
    render(<HomePage />);

    const birthdaysTab = screen.getByText(/Aniversários/i);
    fireEvent.click(birthdaysTab);

    const header = screen.getByText(/Aniversários do mês/i);
    expect(header).toBeInTheDocument();

    expect(screen.getByTestId('birthday-content')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<HomePage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
