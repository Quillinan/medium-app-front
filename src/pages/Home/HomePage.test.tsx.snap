// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-full"
  >
    <nav
      class="bg-gray-800"
      data-testid="navbar"
    >
      <div
        class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8"
      >
        <div
          class="flex h-16 items-center justify-between"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt="Your Company"
                class="h-8 w-8"
                src="https://tailwindui.com/plus/img/logos/mark.svg?color=indigo&shade=500"
              />
            </div>
            <div
              class="hidden md:block"
            >
              <div
                class="ml-10 flex items-baseline space-x-4"
              >
                <button
                  class="rounded-md px-3 py-2 text-sm font-medium bg-gray-900 text-white"
                >
                  Menu
                </button>
                <button
                  class="rounded-md px-3 py-2 text-sm font-medium text-gray-300 hover:bg-gray-700 hover:text-white"
                >
                  Dashboard
                </button>
                <button
                  class="rounded-md px-3 py-2 text-sm font-medium text-gray-300 hover:bg-gray-700 hover:text-white"
                >
                  Aniversários
                </button>
              </div>
            </div>
          </div>
          <div
            class="hidden md:block"
          >
            <div
              class="ml-4 flex items-center md:ml-6"
            >
              <button
                class="relative rounded-full bg-gray-800 p-1 text-gray-400 hover:text-white focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Button
                </span>
                <img
                  alt="User"
                  class="h-8 w-8 rounded-full"
                  src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <header
      class="bg-white shadow"
      data-testid="header"
    >
      <div
        class="mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8"
      >
        <h1
          class="text-3xl font-bold tracking-tight text-gray-900"
        >
          Menu
        </h1>
      </div>
    </header>
    <main
      data-testid="main-content"
    >
      <div
        class="mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8"
      >
        <div>
          Conteúdo da aba Menu
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
